x-pack/platform/plugins/shared/alerting/server/backfill_client/backfill_client.ts

```diff
@@ -97,6 +112,107 @@ export class BackfillClient {
     });
   }
 
+  private async deleteAdHocRunsAndTasks({
+    unsecuredSavedObjectsClient,
+    adHocRuns,
+    shouldUpdateGaps,
+    internalSavedObjectsRepository,
+    eventLogClient,
+    eventLogger,
+    actionsClient,
+  }: {
+    unsecuredSavedObjectsClient: SavedObjectsClientContract;
+    adHocRuns: Array<SavedObjectsFindResult<AdHocRunSO>>;
+    shouldUpdateGaps?: boolean;
+    internalSavedObjectsRepository?: ISavedObjectsRepository;
+    eventLogClient?: IEventLogClient;
+    eventLogger?: IEventLogger;
+    actionsClient?: ActionsClient;
+  }) {
+    if (adHocRuns.length === 0) return;
+
+    // Prepare backfill metadata for gap updates before deleting SOs
+    const backfillsForGapUpdate =
```

Comment:
suggest creating a const like `const canUpdateGaps = shouldUpdateGaps && actionsClient && internalSavedObjectsRepository && eventLogClient` and reusing.